src/Features/LanguageServer/Protocol/Handler/Breakpoints/ValidateBreakableRangeHandler.cs

```diff
@@ -80,7 +80,31 @@ public ValidateBreakableRangeHandler()
             // zero-width range means line breakpoint:
             var breakpointSpan = result.IsLineBreakpoint ? new TextSpan(span.Start, length: 0) : result.TextSpan;
 
-            return ProtocolConversions.TextSpanToRange(breakpointSpan, text);
+            var breakpointRange = ProtocolConversions.TextSpanToRange(breakpointSpan, text);
+
+            // if the breakpoint we get back covers fewer lines than what was requested, then we might be in a situation where
+            // the breakpoint was expanded due to the user typing some code above the placement. For example:
+            //
+            //     $$
+            // BP: Console.WriteLine(1);
+            //
+            // if the user types "int a =" we'll expand the breakpoint, as syntactically its an assigment expression, but then
+            // when they continue to type "1;" we'll get a request for a breakpoint that spans two lines, and then the above
+            // resolve call will shrink it to one. In that case, we prefer to stick to the end of the requested range.
+            if (!result.IsLineBreakpoint && CoversFewerLines(breakpointRange, request.Range))
```

Comment:
>CoversFewerLines [](http://example.com/codeflow?start=44&length=16)

Would this work without the line condition? 
Seems like the problem would be the same regardless of what kind of whitespace is in between the two statements.

Reply:
Are you saying we do this for any shrinking of the breakpoint range? Or would you just check if the breakpoint range at the start and end of the span are for the same thing?

I don't necessarily object, but personally I think this scenario would feel weird if the breakpoint went to the second statement instead of the first:
![BreakpointSpanAdjust2](https://user-images.githubusercontent.com/754264/188032342-66d07b08-5dd1-4aff-8a17-524cb5badd55.gif)

Reply:
Yeah, for any shrinking. I don't see line breaks as special in anyway wrt breakpoint spans. 

If someone has a reason to put multiple statements on the same line then they'd probably not want the original behavior of the breakpoint span moving just because they started typing in front of it either.